EventListener/CsrfListener.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+/*
+ * This file is part of the Symfony framework.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sensio\Bundle\FrameworkExtraBundle\EventListener;
+
+use Sensio\Bundle\FrameworkExtraBundle\Configuration\Csrf;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\HttpKernel\Event\FilterControllerEvent;
+use Symfony\Component\HttpKernel\Exception\BadRequestHttpException;
+use Symfony\Component\HttpKernel\KernelEvents;
+use Symfony\Component\Security\Core\Exception\AccessDeniedException;
+use Symfony\Component\Security\Csrf\CsrfToken;
+use Symfony\Component\Security\Csrf\CsrfTokenManagerInterface;
+
+/**
+ * CsrfListener handles csrf restrictions on actions.
+ *
+ * @author Konstantin Myakshin <koc-dp@yandex.ru>
+ */
+class CsrfListener implements EventSubscriberInterface
+{
+    private $csrfTokenManager;
+
+    public function __construct(CsrfTokenManagerInterface $csrfTokenManager = null)
+    {
+        $this->csrfTokenManager = $csrfTokenManager;
+    }
+
+    public function onKernelController(FilterControllerEvent $event)
+    {
+        $request = $event->getRequest();
+        if (!$configuration = $request->attributes->get('_csrf')) {
+            return;
+        }
+        /* @var $configuration Csrf */
+
+        if (null === $this->csrfTokenManager) {
+            throw new \LogicException('To use the @Csrf tag, you need to install the "symfony/security-csrf".');
+        }
+
+        $token = $request->get($configuration->getParam());
+
+        if (!$this->csrfTokenManager->isTokenValid(new CsrfToken($configuration->getIntention(), $token))) {
+            throw new BadRequestHttpException('Csrf token is invalid.');
```

Comment:
I found examples of both. [Spring returns a 403](http://docs.spring.io/spring-security/site/docs/current/reference/html/csrf.html#csrf-configure):

> The next step is to include Spring Security’s CSRF protection within your application. Some frameworks handle invalid CSRF tokens by invaliding the user’s session, but this causes its own problems. Instead by default Spring Security’s CSRF protection will produce an HTTP 403 access denied.